src/components/PupilJourneyComponents/QuizMCQSingleAnswer/QuizMCQSingleAnswer.tsx

```diff
@@ -1,75 +1,106 @@
+import { QuizEngineContext } from "@/components/PupilJourneyComponents/QuizEngineProvider";
 
 export type QuizMCQSingleAnswerProps = {
   questionUid: string;
   answers: MCAnswer[];
-  currentQuestionIndex: number;
-  isFeedbackMode: boolean;
-  selectedAnswer?: { answer?: MCAnswer | null; index: number };
-  setSelectedAnswer: (val: { answer?: MCAnswer | null; index: number }) => void;
 };
 
 export const QuizMCQSingleAnswer = (props: QuizMCQSingleAnswerProps) => {
-  const {
-    questionUid,
-    currentQuestionIndex,
-    answers,
-    isFeedbackMode,
-    selectedAnswer,
-    setSelectedAnswer,
-  } = props;
+  const { questionUid, answers } = props;
+
+  const quizEngineContext = useContext(QuizEngineContext);
+
+  const currentQuestionIndex = quizEngineContext?.currentQuestionIndex || 0;
+  const questionState = quizEngineContext?.questionState[currentQuestionIndex];
+
+  const [selectedAnswer, setSelectedAnswer] = useState<
+    MCAnswer | null | undefined
+  >(null);
+
+  useEffect(() => {
+    if (questionState?.mode === "grading") {
+      quizEngineContext?.handleSubmitMCAnswer(selectedAnswer);
+    }
+  }, [questionState, currentQuestionIndex, quizEngineContext, selectedAnswer]);
+
+  if (!questionState) {
+    return null;
+  }
+
+  const isFeedbackMode = questionState.mode === "feedback";
 
   return (
     <OakRadioGroup
       name={questionUid || "quiz"}
       $flexDirection={"column"}
       onChange={(e) => {
         const targetIndex = e.target.tabIndex;
-        const selectedAnswer = answers[targetIndex];
-        setSelectedAnswer({
-          answer: selectedAnswer,
-          index: targetIndex,
-        });
+        if (questionState?.mode === "init") {
+          quizEngineContext?.updateQuestionMode("input");
+        }
+        setSelectedAnswer(answers[targetIndex]);
       }}
       disabled={isFeedbackMode}
     >
       {answers?.map((answer, i) => {
         return (
           <OakBox key={`radio-${i}`}>
             {answer.answer.map((answerItem) => {
-              const isSelected = selectedAnswer?.index === i;
-              const incorrectColor = isSelected ? "red" : undefined;
+              let feedbackModeColor: "oakGreen" | "red" | undefined;
 
-              const feedbackModeColor = answer.answer_is_correct
-                ? "oakGreen"
-                : incorrectColor;
+              if (questionState.feedback?.[i] === "correct") {
+                feedbackModeColor = "oakGreen";
+              } else if (questionState.feedback?.[i] === "incorrect") {
+                feedbackModeColor = "red";
+              }
 
               const backgroundColor = isFeedbackMode
                 ? feedbackModeColor
                 : undefined;
 
               const color = backgroundColor ? "text-inverted" : undefined;
 
+              const correctChoice = (
+                <>
+                  {answer.answer_is_correct && isFeedbackMode && (
```

Comment:
Just a question for my own edification but should I only expect to see snake case when something is coming from the backend (I.e. JSON)?

Reply:
Alas only sometimes